tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9043,13 +9028,77 @@ TEST_F(ExecutionTest, BarycentricsTest) {
         int pixelY = (int)((1 - y1) * (height - 1) / 2);
         int offset = pixelSize * (pixelX + pixelY * width) / sizeof(pPixels[0]);
         LogCommentFmt(L"location  %u %u, value %f, %f, %f", pixelX, pixelY, pPixels[offset], pPixels[offset + 1], pPixels[offset + 2]);
-        VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset], w0, tolerance));
-        VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 1], w1, tolerance));
-        VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 2], w2, tolerance));
+        if (!checkOrdering){
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset], w0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 1], w1, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 2], w2, tolerance));
+        }
+        else{
+            // If the ordering constraint is met, then this pixel's RGBA should be all 1.0's
+            // since the shader only returns float4<1.0,1.0,1.0,1.0> when this condition is met.
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset]    , 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 1], 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 2], 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 3], 1.0, tolerance));        
```

Comment:
Am I correct that the old test required looser precision as a result of the changed dimensions?

Reply:
The old test required higher precision because the dimensions were relatively larger.
Now that we want the dimensions to be smaller, it seems the precision should be looser.